frontend/src/pages/EditRolesPage/components/buttons/EditRolesButton.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { Profiler } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Button } from '@SharedComponents';
+import * as ROUTES from '@Constants/routes';
+import {
+  AppUserResponse,
+  AppUserInterviewAvailability,
+  InterviewDatesResponse,
+} from '@Services/api';
+import {
+  CssBaseline,
+  Drawer,
+  AppBar,
+  Toolbar,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+  Divider,
+  IconButton,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Hidden,
+} from '@material-ui/core';
+import { updateUserById } from '@Services/UserService';
+import { AppUserPostRequest } from '@Services/api';
+
+interface EditRolesButtonProps {
+  id: number;
+  role: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+}
+interface INITIAL_STATES {
+  isPopupOpen?: boolean;
+  id?: number;
+  role?: string;
+  email?: string;
+  firstName?: string;
+  lastName?: string;
+}
+
+class EditRolesButton extends React.Component<
```

Comment:
I think we can just make this a functional component? Do we use class components anywhere else in the codebase? (sorry it's been a while)

Reply:
We use a mix of class and functional components in the codebase I believe. In this case it's fine for it to be a functional component. I'll go ahead and make that change accordingly.